modules/custom/d_p_reference_content/src/Helpers/ContentHelper.php

```diff
@@ -148,4 +148,37 @@ private function prepareFieldValues($data) {
     return $result;
   }
 
+  /**
+   * Delete unpublished content.
+   *
+   * @param array $values
+   *   Array with content values.
+   *
+   * @return array
+   *   Content without unpublished nodes.
+   */
+  public function getPublishedContent(array $values) {
+    $nids = [];
+    foreach ($values as $value) {
+      $nids[] = $value['target_id'];
+    }
+
+    $results = $this->connection->select('node_field_data', 'nfd')
+      ->fields('nfd', ['nid'])
+      ->condition('nfd.status', 1)
```

Comment:
A czy nie dałoby się dodać warunku status=1 do funkcji `getSortedContentByType`? Bo rozumiem, że tu pobieramy nids, a potem w kolejnej funkcji usuwamy te nieopublikowane?

Reply:
Nie, to są pobierane rzeczy jako automatyczne value. W takim wypadku ręcznie wybrane elementy które nie są opublikowane nadal by się wyświetlały.